File: Level 2/쿼드압축 후 개수 세기/solution.py
def solution(arr):
    answer = [0, 0]

    quadtree(0, 0, len(arr), arr, answer)

    return answer


def quadtree(x, y, n, arr, answer):
    for i in range(x, x + n):
        for j in range(y, y + n):
            if arr[x][y] != arr[i][j]:
                n //= 2
                quadtree(x, y, n, arr, answer)
                quadtree(x + n, y, n, arr, answer)
                quadtree(x, y + n, n, arr, answer)
                quadtree(x + n, y + n, n, arr, answer)
                return

    answer[arr[x][y]] += 1

File: Level 2/쿼드압축 후 개수 세기/solution.js
function solution(arr) {
  const answer = [0, 0];

  quadtree(0, 0, arr.length, arr, answer);

  return answer;
}

function quadtree(x, y, n, arr, answer) {
  for (let i = x; i < x + n; i++) {
    for (let j = y; j < y + n; j++) {
      if (arr[x][y] !== arr[i][j]) {
        n = Math.floor(n / 2);
        quadtree(x, y, n, arr, answer);
        quadtree(x + n, y, n, arr, answer);
        quadtree(x, y + n, n, arr, answer);
        quadtree(x + n, y + n, n, arr, answer);
        return;
      }
    }
  }

  answer[arr[x][y]]++;
}
